Refactor/harsh discipline

oisinod opened the pull request:
### Description
Harsh Discipline uses atonement normalizer and not event anymore
closes #5691

trevorm4 commented on src/analysis/retail/priest/discipline/modules/spells/HarshDiscipline.tsx:
```diff
@@ -51,16 +54,20 @@ class HarshDiscipline extends Analyzer {
     }
 
-    this.harshAtonement += event.healEvent.amount;
+    if (damageEvent.penanceBoltNumber + 1 <= this.expectedBolts) {
```
```suggestion
    if (damageEvent.penanceBoltNumber < this.expectedBolts) {
```

trevorm4 commented on src/analysis/retail/priest/discipline/modules/spells/HarshDiscipline.tsx:
```diff
@@ -25,6 +22,7 @@ class HarshDiscipline extends Analyzer {
   private harshAtonement = 0;
   private harshDirect = 0;
   private harshPenances = 0;
+  private testAtonement = 0;
```
```suggestion
```

oisinod commented:
@trevorm4 I've added another commit adressing the comments you had but since they were missing I added the data for this module to also work with shadow penance

Thanks!
